File: Week-Wise-Assignments/Assignment 7/Compressed Solutions/Python Programs/5_linear_search.py
def linear_search(arr, target):
    for i in range(len(arr)):
        if arr[i] == target:
            return i
    return -1

n = int(input("Enter size: "))
arr = list(map(int, input("Enter elements: ").split()))
target = int(input("Enter target: "))

result = linear_search(arr, target)
if result != -1:
    print(f"Found at index {result}")
else:
    print("Not found")

File: Week-Wise-Assignments/Assignment 7/Compressed Solutions/C Programs/5_linear_search.c
#include <stdio.h>

int linearSearch(int arr[], int n, int target) {
    for (int i = 0; i < n; i++) {
        if (arr[i] == target) return i;
    }
    return -1;
}

int main() {
    int n, target;
    printf("Enter size: ");
    scanf("%d", &n);
    int arr[n];
    
    printf("Enter elements: ");
    for (int i = 0; i < n; i++) scanf("%d", &arr[i]);
    
    printf("Enter target: ");
    scanf("%d", &target);
    
    int result = linearSearch(arr, n, target);
    if (result != -1) {
        printf("Found at index %d\n", result);
    } else {
        printf("Not found\n");
    }
    
    return 0;
}